utils/docker/build-image.sh

```diff
@@ -1,89 +1,194 @@
+      echo -e "\nERROR: '${image_type}' not supported! Use 'minimal' or 'develop'."
+      exit 99
+  else
+      IMAGE_TYPE="${image_type}"
+  fi
+
+  if [[ "${local_copy}" = "true" && ( -n "${repo_url}" || -n "${repo_branch}" ) ]]; then
+    usage_error "--local-copy is mutually exclusive with repository url and/or branch"
+  fi
+
+  if [[ "${local_copy}" = "true" && ( -z "${IMAGE}" && -z "${IMAGE_REPOSITORY}" && -z "${IMAGE_OWNER}" ) ]]; then
+    echo "Sorry.  For local mode you need to specify an --image-owner or an --image-repository or an --image" >&2
+    exit 1
+  fi
+
+  if [[ "${local_copy}" == "true" ]]; then
+    echo "Buildind an image using the current local project state" >&2
+    echo "=*=*=*=*=*=* WARNING: Your Docker image could include uncommitted changes! *=*=*=*=*=*=" >&2
+  else
+    echo "Building an image using a wft4galaxy pulled from a git repository" >&2
+  fi
+}
+
+function cd_to_project_root() {
+  local src_dir_root="${script_path}/../../"
+  # Try to verify that we have the right directory
+  if [[ ! -d "${src_dir_root}/wft4galaxy" || ! -d "${src_dir_root}/utils" ]]; then
+    error "I think the wft4galaxy root directory should be '${src_dir_root}' but I can't \n" \
+          "find the 'wft4galaxy' and 'utils' subdirectories there. Aborting!"
+  fi
+
+  cd "${src_dir_root}"
+}
+
+function build_local_image() {
+  if [[ -z "${IMAGE}" && -z "${IMAGE_REPOSITORY}" && -z "${IMAGE_OWNER}" ]]; then
+    echo "BUG! Trying to build a local image without image information" >&2
+    exit 2
+  fi
+
+  # assemble any missing image info
+  source ${script_path}/set-docker-image-info.sh
+
+  cd_to_project_root
+  docker build . -f "${script_path}/${IMAGE_TYPE}/Dockerfile.local" -t ${IMAGE}
+}
+
+function build_repo_image() {
+  # set git && image info
+  source ${script_path}/set-git-repo-info.sh ${repo_url} ${repo_branch}
+  source ${script_path}/set-docker-image-info.sh
+
+  cd_to_project_root
+  docker build . -f "${script_path}/${IMAGE_TYPE}/Dockerfile.git" \
+		     --build-arg git_branch=${GIT_BRANCH} --build-arg git_url=${GIT_HTTPS} -t ${IMAGE}
+}
+
+########## main ###########
 
 # absolute path of the current script
 script_path="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-# absolute path of the image folder
-image_path="${script_path}/${image_type}"
-# check whether the image type exists
-if [[ ! -d ${image_path} ]]; then
-    echo -e "\nThe image type '${image_type}' doen't exist!!!"
-    exit 99
-fi
-
-# set git && image info
-source ${script_path}/set-git-repo-info.sh ${repo_url} ${repo_branch}
-source ${script_path}/set-docker-image-info.sh
+# initialize variables to default values, if any
+IMAGE=""
+IMAGE_REPOSITORY=""
+IMAGE_REGISTRY=""
+IMAGE_OWNER=""
+IMAGE_NAME=""
+IMAGE_TAG=""
+IMAGE_TYPE="minimal"
+repo_url=""
+repo_branch=""
+local_copy="false"
 
-# Need to cd into this script's directory because image-config assumes it's running within it
-cd "${image_path}" > /dev/null
+parse_args "${@}"
```

Comment:
Same ``unbound variable`` error in https://github.com/phnmnl/wft4galaxy/pull/44/files#diff-b1353c1dd16b97604e8551563155ca11R110